src/network/connectionManager.ts

```diff
@@ -0,0 +1,1292 @@
+import toast from 'react-hot-toast'
+
+// Global timeout on pending commands, it will be bad if we hit this case.
+const PENDING_COMMAND_TIMEOUT = 60_000
+
+export class ConnectionManager extends EventTarget {
+  started: boolean
+  inSequence = 1
+  _camControlsCameraChange = () => {}
+  id: string
+
+  /**
+   * The pendingCommands object is a map of the commands that have been sent to the engine that are still waiting on a reply
+   */
+  pendingCommands: {
+    [commandId: string]: PendingMessage
+  } = {}
+
+  /**
+   * A map of the responses to the WASM, this response map allow
+   * us to look up the response by command id
+   */
+  responseMap: ResponseMap = {}
+  /**
+   * A counter that is incremented with each command sent over the *unreliable* channel to the engine.
+   * This is compared to the latest received {@link inSequence} number to determine if we should ignore
+   * any out-of-order late responses in the unreliable channel.
+   */
+  outSequence = 1
+  commandLogs: CommandLog[] = []
+
+  connection: Connection | undefined
+  sceneInfra: SceneInfra | undefined
+  codeManager: CodeManager | undefined
+  kclManager: KclManager | undefined
+
+  // Circular dependency that is why it can be undefined
+  rustContext: RustContext | undefined
+
+  streamDimensions = {
+    // startFromWasm uses 256, 256
+    width: 256,
+    height: 256,
+  }
+  settings: SettingsViaQueryString = {
+    pool: null,
+    theme: Themes.Dark,
+    highlightEdges: true,
+    enableSSAO: true,
+    showScaleGrid: false,
+    cameraProjection: 'orthographic', // Gotcha: was perspective now is orthographic
+    cameraOrbit: 'spherical',
+  }
+
+  subscriptions: {
+    [event: string]: {
+      [localUnsubscribeId: string]: (a: any) => void
+    }
+  } = {} as any
+  unreliableSubscriptions: {
+    [event: string]: {
+      [localUnsubscribeId: string]: (a: any) => void
+    }
+  } = {} as any
+  _commandLogCallBack: (command: CommandLog[]) => void = () => {}
+  // Rogue runtime dependency from the modeling machine. hope it is there!
+  modelingSend: ReturnType<typeof useModelingContext>['send'] =
+    (() => {}) as any
+  // Any event listener into this map to be cleaned up later
+  // helps avoids duplicates as well
+  allEventListeners: Map<string, IEventListenerTracked>
+
+  callbackOnUnitTestingConnection: (() => void) | null
+
+  constructor(settings?: SettingsViaQueryString) {
+    super()
+    this.started = false
+    this.allEventListeners = new Map()
+    this.id = uuidv4()
+    this.callbackOnUnitTestingConnection = null
+
+    if (settings) {
+      // completely overwrite the settings
+      this.settings = settings
+    } else {
+      // Gotcha: This runs in testing environments
+      this.settings = {
+        pool: null,
+        theme: Themes.Dark,
+        highlightEdges: true,
+        enableSSAO: true,
+        showScaleGrid: false,
+        cameraProjection: 'perspective',
+        cameraOrbit: 'spherical',
+      }
+    }
+  }
```

Comment:
I see how you rewrote this from https://github.com/KittyCAD/modeling-app/blob/main/src/lang/std/engineConnection.ts#L1431-L1443 to make it more explicit, but it kinda seems like a default value for `settings` so could it be reflected in the constructor arguments as such maybe?

Reply:
Ah nvm I see your comments on L163-164, we are handling settings weird don't worry about what I suggested above.